File: src/test/java/io/tripled/social/client/presentation/controller/FollowControllerTest.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.SpiedFollowUserUseCase;
import io.tripled.social.client.presentation.TestOutput;
import io.tripled.social.client.presentation.request.FollowCliRequest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

class FollowControllerTest {

  private FollowController followController;
  private SpiedFollowUserUseCase service;

  @BeforeEach
  void setUp() {
    service = new SpiedFollowUserUseCase();
    followController = new FollowController(service);
  }

  @Test
  void executesCommand() {
    FollowCliRequest request = new FollowCliRequest("Charlie", "Alice");

    followController.execute(request, new TestOutput());

    assertThat(service.userName, equalTo("Charlie"));
    assertThat(service.userNameToFollow, equalTo("Alice"));
  }

}

File: src/test/kotlin/io/tripled/social/client/presentation/controller/FollowControllerTest.kt
package io.tripled.social.client.presentation.controller

import io.tripled.social.client.application.SpiedFollowUserUseCase
import io.tripled.social.client.presentation.TestOutput
import io.tripled.social.client.presentation.request.FollowCliRequest
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

internal class FollowControllerTest {
    private val service: SpiedFollowUserUseCase = SpiedFollowUserUseCase()
    private val followController: FollowController = FollowController(service)

    @Test
    fun executesCommand() {
        val request = FollowCliRequest("Charlie", "Alice")

        followController!!.execute(request, TestOutput())

        MatcherAssert.assertThat<String?>(service!!.userName, Matchers.equalTo<String?>("Charlie"))
        MatcherAssert.assertThat<String?>(service!!.userNameToFollow, Matchers.equalTo<String?>("Alice"))
    }
}
